source/connection/connection-pools.txt

```diff
@@ -0,0 +1,160 @@
+value of maxPoolSize, the next request to that server will wait until a 
+connection becomes available.
+
+Each ``MongoClient`` instance opens two more sockets per server in your MongoDB 
+topology for monitoring the server's state.
+
+Configure a Connection Pool
+---------------------------
+
+You can specify settings for your connection pool using either a connection
+string or by passing a ``MongoClientSettings`` object to the 
+``MongoClients.create()`` method.
+
+The following code creates a client with a maximum connection pool size of ``50``. 
+Select the :guilabel:`Connection String` or :guilabel:`MongoClientSettings`  tab to
+see the corresponding syntax:
+
+.. tabs::
+
+  .. tab:: Connection String
+    :tabid: uri
+    
+    .. code-block:: java
+
+      ConnectionString connectionString = "mongodb://<host>:<port>/?maxPoolSize=50"
+      MongoClient mongoClient = MongoClients.create(connectionString)
+
+    The following are connection string settings you can use to configure your 
+    connection pool:
+
+    .. list-table::
+      :widths: 25,75                                                          
+      :header-rows: 1
+
+      * - Setting
+        - Description
+        
+      * - ``connectTimeoutMS``
+
+        - Specifies the maximum amount of time, in milliseconds, the Java driver 
+          waits for a connection to open before timing out. A value of 0 instructs 
+          the driver to never time out while waiting for a connection to open.
+          
+          *Default:* ``10000`` (10 seconds)
+      
+      * - ``maxConnecting``
+      
+        - Maximum number of connections a pool may establish
+          concurrently.
+
+          *Default:* ``2``
+      
+      * - ``maxIdleTimeMS``
+      
+        - The maximum number of milliseconds that a connection can 
+          remain idle in the pool before being removed and closed.
+
+          *Default:* ``0``
+      
+      * - ``maxPoolSize``
+
+        - Maximum number of connections opened in the pool. When the 
+          connection pool reaches the maximum number of connections, new 
+          connections wait up until to the value of 
+          ``waitQueueTimeoutMS``.
+
+          *Default:* ``100`` 
+
+      * - ``minPoolSize``
+
+        - Minimum number of connections opened in the pool. 
+          The value of ``minPoolSize`` must be less than 
+          the value of ``maxPoolSize``.
+
+          *Default*: ``0``
+
+      * - ``socketTimeoutMS``
+
+        - Number of milliseconds to wait before timeout on a TCP 
+          connection.
+          
+          Do *not* use ``socketTimeoutMS`` as a mechanism for 
+          preventing long-running server operations.
+
+          Setting low socket timeouts may result in operations that error 
+          before the server responds.
+          
+          *Default*: ``0``, which means no timeout.
+
+      * - ``waitQueueTimeoutMS``
+
+        - Maximum wait time in milliseconds that a thread can wait for 
```

Comment:
In the reactive streams driver, it's not actually a thread that is waiting.  Consider replacing "thread" with "operation".